eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -12,16 +12,40 @@ parameters:
   default: []
 - name: JobTemplatePath
   type: string
+# Set this to false to do a full checkout for private repositories with the azure pipelines service connection
+- name: SparseCheckout
+  type: boolean
+  default: true
+- name: SparseCheckoutPaths
+  type: object
+  default: []
+- name: Pool
+  type: string
+  default: azsdk-pool-mms-ubuntu-1804-general
+- name: OsVmImage
+  type: string
+  default: MMSUbuntu18.04
 
 jobs:
 - job: generate_matrix
   variables:
     displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
   pool:
-    name: Azure Pipelines
-    vmImage: ubuntu-18.04
+    name: ${{ parameters.Pool }}
```

Comment:
Is there a reason to parameterize this?

Reply:
The reason I have it this way is I want the `/eng/common/scripts/job-matrix/samples/matrix-test.yml` to be a working sample (partly because non-working samples are bad, and because I use it for testing changes like these). Since I test this sample pipeline in the playground project, it doesn't have access to the 1es pool, hence why I need to override the pool/image.